### PR TITLE
Refactor for android change.

### PR DESCRIPTION
**Pull request by tanin47**
- Move some instance methods to module methods. Because instance methods are difficult to test (e.g. require mocking the whole class, especially its constructor).
- Remove reading gn args from downloaded build. We have args.gn on the testcase details page for more than 2 months now.

### CODE REVIEW

File: tool/clusterfuzz/binary_providers.py [hunk 1]
```diff
@@ -137,15 +137,6 @@ def gclient_runhooks_msan(source_dir, msan_track_origins):
   )
 
 
-def read_gn_args(gn_args, downloaded_args_gn_path):
```
Comment (tanin47) [inline tool/clusterfuzz/binary_providers.py]: We don't use this anymore. Args.gn has been on the testcase detail page for a few months now.

File: tool/clusterfuzz/binary_providers.py [hunk 2]
```diff
@@ -157,6 +148,55 @@ def setup_gn_goma_params(goma_dir, gn_args):
...
+    args.append('%s = %s' % (key, val))
+  return '\n'.join(args)
+
+
+def download_build(dest, url, binary_name):
+  """Download and extract a build (if it's not already there)."""
+  if os.path.exists(dest):
+    return dest
+
+  logger.info('Downloading build data...')
```
Comment (Dor1s) [inline tool/clusterfuzz/binary_providers.py]: Would it be useful to write something like:
```python
'Downloading build data into %s.' % common.CLUSTERFUZZ_BUILDS_DIR
```

to give a better understanding of the tool activity to the developer?

File: tool/clusterfuzz/binary_providers.py [hunk 3]
```diff
@@ -157,6 +148,55 @@ def setup_gn_goma_params(goma_dir, gn_args):
...
+  common.gsutil('cp %s .' % gsutil_path, common.CLUSTERFUZZ_CACHE_DIR)
+
+  filename = os.path.basename(gsutil_path)
+  saved_file = os.path.join(common.CLUSTERFUZZ_CACHE_DIR, filename)
+
+  common.execute(
+      'unzip', '-q %s -d %s' % (saved_file, common.CLUSTERFUZZ_BUILDS_DIR),
+      cwd=common.CLUSTERFUZZ_DIR)
+
+  logger.info('Cleaning up...')
```
Comment (Dor1s) [inline tool/clusterfuzz/binary_providers.py]: The same question, maybe print the path here?

File: tool/clusterfuzz/binary_providers.py [hunk 4]
```diff
@@ -157,6 +148,55 @@ def setup_gn_goma_params(goma_dir, gn_args):
...
+
+  logger.info('Downloading build data...')
+  common.ensure_dir(common.CLUSTERFUZZ_BUILDS_DIR)
+
+  gsutil_path = url.replace(
+      'https://storage.cloud.google.com/', 'gs://')
+  common.gsutil('cp %s .' % gsutil_path, common.CLUSTERFUZZ_CACHE_DIR)
+
+  filename = os.path.basename(gsutil_path)
+  saved_file = os.path.join(common.CLUSTERFUZZ_CACHE_DIR, filename)
```
Comment (Dor1s) [inline tool/clusterfuzz/binary_providers.py]: Do we clean up the cache dir?

**Pull request merged**